WebDevStudios/WDS-Blocks
# Feature/users grid

### DESCRIPTION ###
Creates a Users Grid block, inspired by this tweet: https://twitter.com/melchoyce/status/989664968818741248
![](https://dl.dropbox.com/s/zcqvnxorrzzhfgd/Screenshot%202018-04-30%2015.53.36.jpg?dl=0)

This also breaks out the Search Component into its own standalone component apart from Related Posts, adding customization to it so that it can be used to search for posts, pages, users, post types, etc.

With that, a lot of files are moved out of Related Posts leaving just:
```
editor.scss
index.js
render.php
style.scss
```

Everything else was able to be moved into `/components/search-edit-component/` directory. I am 100% okay if we want to rename that - I couldn't think of a snappy name for the thing that is the entire block of the relationship/search/select block.

The block itself is much like Related Posts - it provides a field for a title and a search field for users to search for an select existing Users to display in a grid.

Currently, the WP-API only displays users who have published posts which threw me for a loop for a bit.  If you create a user and they do not have a published post, they are not able to be searched or displayed via the API at the present time.

This also fixes an issue I guess we missed, where when Related Posts are added they may not always display in the right order in the backend depending on the order they were published.  They were still being rendered in publish order rather than the order they were listed in the right column, but this has (I believe) been fixed by adding `orderby=include` to our WP-API queries so that the posts/users/etc. are displayed in the order in which they are added and exist in the array.

### SCREENSHOTS ###
Backend:
![](https://dl.dropbox.com/s/ctmt1ll3a62bbbs/Screenshot%202018-04-30%2015.18.53.jpg?dl=0)
![](https://dl.dropbox.com/s/zugsmuziscf9omr/Screenshot%202018-04-30%2015.25.52.jpg?dl=0)
![](https://dl.dropbox.com/s/othj1a041w0jwfd/Screenshot%202018-04-30%2015.26.16.jpg?dl=0)

Frontend:
![](https://dl.dropbox.com/s/fr3vusra9smz6a1/Screenshot%202018-04-30%2015.26.32.jpg?dl=0)

As with our other blocks, this allows for the user of our background/text/other options:
![](https://dl.dropbox.com/s/5yunyfb2nrk6mbh/Screenshot%202018-04-30%2015.26.53.jpg?dl=0)

### OTHER ###
- [ ] Does this issue pass linting? (ESLint)

### STEPS TO VERIFY ###
Checkout the branch and add the `Users Grid (WDS)` block to your page. Search for users, add them, remove them, save them, do the things!  Also, add the `Related Posts (WDS)` block to your page and ensure that this still works as expected.

_**Assigning to Greg and Jo to review as this builds on a lot of code Jo wrote, so I'd like to have him run through and review as well if possible.**_

### DOCUMENTATION ###
Will this pull request require updating the WDS Blocks [wiki](https://github.com/WebDevStudios/wds-blocks/wiki)?
Yes - I'll add to the docs if/when this gets approved and merged.
